src/v/cloud_storage/tests/remote_partition_test.cc

```diff
@@ -1077,6 +1092,32 @@ FIXTURE_TEST(test_remote_partition_read_cached_index, cloud_storage_fixture) {
           = reader.consume(test_consumer(), model::no_timeout).get();
         BOOST_REQUIRE(!headers_read.empty());
     }
+
+    /// Test that the reader respects the deadline
+    {
+        partition_probe probe(manifest.get_ntp());
+        auto manifest_view = ss::make_shared<async_manifest_view>(
+          api, cache, manifest, bucket_name, path_provider);
+        auto partition = ss::make_shared<remote_partition>(
+          manifest_view, api.local(), cache.local(), bucket_name, probe);
+        auto partition_stop = ss::defer(
+          [&partition] { partition->stop().get(); });
+        partition->start().get();
+
+        cloud_log_reader_config reader_config(
+          model::offset_cast(base), model::offset_cast(max));
+        reader_config.start_offset = model::offset_cast(
+          segments.front().base_offset);
+        reader_config.max_bytes = max_bytes_limit;
+        vlog(test_log.info, "read last segment: {}", reader_config);
+        auto reader = partition->make_reader(reader_config).get().reader;
+        auto headers_read
+          = reader.consume(slow_consumer(), model::timeout_clock::now() + 10ms)
+              .get();
+        // We expect the consumer to consume the first batch, then stall
+        // then consume the second batch and bail out due to timeout.
+        BOOST_REQUIRE_EQUAL(headers_read.size(), 2);
```

Comment:
should do this twice in the test where we use either a different deadline or a non-slow consumer so we can show the same code with a slightly different setting gives a _different_ headers_read.size() outcome?